File: 2154KeepMultiplyingFoundValuesbyTwo/KeepMultiplyingFoundValuesByTwo.java
import java.util.LinkedHashMap;

class KeepMultiplyingFoundValuesByTwo {
      private class Solution{
        public int findFinalValue(int[] nums, int original) {
           LinkedHashMap<Integer,Integer> map=new LinkedHashMap<Integer,Integer>();
          for(int num:nums){
            map.put(num,1);
          }
          while(true){
            if(!map.containsKey(original)){
              return original;
            }
            original*=2;
          }

        }
      }
}

File: 2154KeepMultiplyingFoundValuesbyTwo/KeepMultiplyingFoundValuesByTwo.js
/**
 * @param {number[]} nums
 * @param {number} original
 * @return {number}
 */
var findFinalValue = function(nums, original) {
    let hash_map={};
    for(let num in nums){
        hash_map[num]=1
    }

    while(true){
        if(!original.toString() in hash_map){
            return original;
        }
        original*=2

    }
};